src/core/stack_switching/stack_switching.mjs

```diff
@@ -11,29 +11,38 @@ import {
 } from "./create_invokes.mjs";
 import { initSuspenders } from "./suspenders.mjs";
 
-export { promisingApply, createPromising } from "./suspenders.mjs";
-
+export {
+  promisingApply,
+  createPromising,
+  validSuspender,
+  suspenderGlobal,
+} from "./suspenders.mjs";
+export { StackState } from "./stack_state.mjs";
 export { jsWrapperTag };
 
-Module.jspiSupported = false;
-Module.validSuspender = { value: 0 };
+let canConstructWasm = true;
+try {
+  // Check that WebAssembly.Module constructor works -- if dynamic eval is
+  // disabled it might raise.
+  // This is the smallest valid wasm module -- only the magic number and wasm
+  // version.
+  new WebAssembly.Module(new Uint8Array([0, 97, 115, 109, 1, 0, 0, 0]));
+} catch (e) {
+  canConstructWasm = false;
+}
 
 // wasm-feature-detect uses `"Suspender" in WebAssembly` feature detect JSPI. It
 // is not 100% clear based on the text of the JSPI proposal that this will
 // actually work in the future, but if it breaks we can replace it with
 // something else that does work.
-if ("Suspender" in WebAssembly) {
-  try {
-    // Check that WebAssembly.Module constructor works -- if dynamic eval is
-    // disabled it might raise.
-    // This is the smallest valid wasm module -- only the magic number and wasm
-    // version.
-    new WebAssembly.Module(new Uint8Array([0, 97, 115, 109, 1, 0, 0, 0]));
-    Module.jspiSupported = true;
-  } catch (e) {}
-}
+export const newJspiSupported = canConstructWasm && "Suspending" in WebAssembly;
+export const oldJspiSupported = canConstructWasm && "Suspender" in WebAssembly;
```

Comment:
Let's mention the EOL (M128) of the old jspi, so we can know when to remove the old jspi.

Reply:
It should be the end of life of node v22.

Reply:
I see. That's a long way off, then.

Reply:
We can perhaps drop support sooner, say after node 24 comes out.

Reply:
Sure, I'm okay with aggressively dropping backward compatibility if necessary.